### PR TITLE
Moved source to site distance calculation to qcore

### PR DESCRIPTION
**Pull request by sungeunbae**
As this is not IM calculation specific, and used by other repos (eg. Empirical Engine), it is to be relocated to qcore

Other PRs to follow from qcore, Empirical Engine and slurm_gm_workflow.

### CODE REVIEW

**Review by joelridden — COMMENTED** [review]
Just needs the merge from main (this will fix the jenkins issue), but should wait for the other PR's to go through before we approve this otherwise rrup can't be calculated in the meantime.

**Review by claudio525 — DISMISSED** [review]

**Review by lispandfound — DISMISSED** [review]

**Review by AndrewRidden-Harper — APPROVED** [review]
looks good to me, noting Joel's comment

**Review by joelridden — APPROVED** [review]
Just waiting now for the rrup to be added to qcore, but will still approve to make it easy once that's done

**Review by lispandfound — APPROVED** [review]

**Review by AndrewRidden-Harper — APPROVED** [review]

**Pull request merged**
